Runtime 8.9.0 (#2754)
astronomer/docs: astro/runtime-release-notes.md

To upgrade Astro Runtime, see [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, see [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

> ## Astro Runtime 8.9.0

Author: Does the chronological order of this page create any confusion about Runtime versions/Airflow versions?

Reviewer: It was discussed on Slack before, our engineers like it better when we organize by versions. It makes it quick to parse for the latest patch of any given version

- Release date: August 28, 2023
- Airflow version: 2.6.3

### Early access Airflow bug fixes

> - Fixed an issue in how a task instance's `log_url` was generated by Airflow to use string concatenation instead of `urljoin`. ([#33063](https://github.com/apache/airflow/pull/33063)) 

Reviewer: ```suggestion
- Fixed an issue where Airflow incorrectly used `urljoin`  to concatenate the string for `log_url`. ([#33063](https://github.com/apache/airflow/pull/33063)) 
```
For a lot of the early access bug fixes, I tend to use the exact copy from the original PR title, even if it might not be perfect grammar or style. It makes it easier to identify specific PRs and reduces the risk of us incorrectly documenting some of the more nuanced technical changes (which there are a lot of for backported bug fixes)

### Additional improvements

> - Upgraded `astronomer-providers` to 1.17.3

Reviewer: ```suggestion
- Upgraded `astronomer-providers` to 1.17.3. See the [`astronomer-providers` changelog](https://github.com/astronomer/astronomer-providers/blob/main/CHANGELOG.rst#1173-2023-08-07) for a complete list of changes.
```

> - Upgraded `astro-sdk-python` to 1.6.2

Reviewer: ```suggestion
- Upgraded `astro-sdk-python` to 1.6.2. See the [Astro Python SDK changelog](https://astro-sdk-python.readthedocs.io/en/stable/CHANGELOG.html#id1) for a complete list of changes.
```

### Bug fixes

> - Fixed an issue so that users can now add and edit DAG Notes or Task notes in Astro Cloud.
> - Fixed trigger logs on AWS

Reviewer: ```suggestion
- Fixed an issue where you could not set DAG or task notes in the Airflow UI for environments running on Astro.
- Fixed an issue with trigger logs for Deployments running on AWS
```

- Fixed an issue where you could not set DAG or task notes in the Airflow UI for environments running on Astro.

## Astro Runtime 8.9.0

- Release date: August 28, 2023
- Airflow version: 2.6.3

### Early access Airflow bug fixes

> - Fixed an issue where Airflow incorrectly used `urljoin`  to concatenate the string for `log_url`. ([#33063](https://github.com/apache/airflow/pull/33063)) 

Reviewer: ```suggestion
- Fixed an issue where Airflow incorrectly used `urljoin` to concatenate the string for `log_url`. ([#33063](https://github.com/apache/airflow/pull/33063)) 
```
